src/vulkan/buffer_descriptor.h

```diff
@@ -67,6 +67,7 @@ class BufferDescriptor : public Descriptor {
   }
 
   std::unique_ptr<Buffer> buffer_;
+  std::vector<std::unique_ptr<Buffer>> not_destroyed_buffers_;
```

Comment:
This seems excessive.
Isn't it better to destroy the old buffer when you are resizing it?  That way you don't accumulate more GPU resources over time.

Reply:
See the comment below